kola/tests/rpmostree/status.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rpmostree
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform"
+)
+
+func init() {
+	register.Register(&register.Test{
+		Run:         rpmOstreeStatus,
+		ClusterSize: 1,
+		Name:        "rhcos.rpmostree.status",
+		Distros:     []string{"rhcos"},
+	})
+}
+
+// hard code the osname for RHCOS
+// TODO: should this also support FCOS?
+var rhcosOsname string = "rhcos"
+
+// Regex to match the likes of "rhcos:openshift/3.10/x86_64/os"
+//
+// Groups:
+// 0 - rhcos
+// 1 - openshift
+// 2 - 3.10
+// 3 - x86_64
+// 4 - os
+var rhcosOriginRegex string = `^(\w+):(\w+)/(\d\.\d+)/(\w+)/(\w+)`
```

Comment:
Let's please not hardcode this - not just because we'd like these tests to work on FCOS too, but also because doing it here will make it *extremely* painful to change - and I'd like to change RHCOS to use a [custom origin](https://github.com/ashcrow/pivot/pull/12) for the initial state too.

If this landed as is, then in order to change RHCOS we'd have to have a patch ready for both this repo and the config, then build an assembler container containing this change, which we'd then have to carefully only use for the *new* OS version.

Basically if one has separate git repos that interlock on each other, changes become extremely hard.

Reply:
Got it; will remove this in an updated commit